File: python/Problem206.py
# Problem: 204. Count Primes
# URL: https://leetcode.com/problems/count-primes/
# Difficulty: Medium
# Tags: Prime

from typing import Optional

# Definition for singly-linked list.
class ListNode:
  def __init__(self, val=0, next=None):
    self.val = val
    self.next = next

class Solution:
  def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
    newHead: Optional[ListNode] = None
    tmp: Optional[ListNode] = None
    while head != None:
      tmp = head.next
      head.next = newHead
      newHead = head
      head = tmp
    return newHead

File: cpp/Problem206.cpp
/*
 * Problem: 206. Reverse Linked List
 * URL: https://leetcode.com/problems/reverse-linked-list/
 * Difficulty: Easy
 * Tags: Linked List
 */

struct ListNode {
	int val;
	ListNode *next;
	ListNode() : val(0), next(nullptr) {}
	ListNode(int x) : val(x), next(nullptr) {}
	ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
	ListNode* reverseList(ListNode* head) {
		ListNode* newHead = nullptr;
		ListNode* tmp;
		while (head != nullptr) {
			tmp = head->next;
			head->next = newHead;
			newHead = head;
			head = tmp;
		}
		return newHead;
	}
};